src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleById, fetchArticleComments } from "../utils/api";
+import Header from "./Header";
+import Loading from "../components/Loading";
+import CommentsList from "./CommentsList";
+import VotesAndCommentCount from "./VotesAndCommentCount";
+
+const SingleArticle = () => {
+  const { article_id } = useParams();
+  const [singleArticle, setSingleArticle] = useState({});
+  const [articleComments, setArticleComments] = useState([]);
+  const [isArticleLoading, setIsArticleLoading] = useState(true);
+
+  useEffect(() => {
+    fetchArticleById(article_id)
+      .then((data) => {
+        setIsArticleLoading(false);
+        setSingleArticle(data);
+      })
+      .catch((err) => console.log(err));
+  }, [article_id]);
+
+  useEffect(() => {
+    fetchArticleComments(article_id)
+      .then((commentsData) => {
+        setIsArticleLoading(false);
```

Comment:
If the comments come first here you will have a page with just the comments and no article, this wouldn't look great from a user perspective, it might make sense to abstract the getting of the comments into it's own component and give it it's own loading state